packages/dart_code_builder/src/to_code.candy

```diff
@@ -0,0 +1,411 @@
+    if body is Some {
+      s = "{s} {(body.unwrap() as ToCode).toCode()}"
+    }
+    if body.map<Bool>({ it is InlineBody }).orElse({ true }) {
+      s = "{s};"
+    }
+    s
+  }
+}
+impl Parameter: ToCode {
+  fun toCode(): String {
+    mut let s = name
+    if type is Some {
+      s = "{(type.unwrap() as ToCode).toCode()} {s}"
+    }
+    if defaultValue is Some {
+      s = "{s} = {(defaultValue.unwrap() as ToCode).toCode()}"
+    }
+    s
+  }
+}
+impl Field: ToCode {
+  fun toCode(): String {
+    mut let s = name
+    if !(mutability is Var) {
+      s = "{(mutability as ToCode).toCode()} {s}"
+    }
+    if isStatic {
+      s = "static {s}"
+    }
+    if initialValue is Some {
+      s = "{s} = {(initialValue.unwrap() as ToCode).toCode()}"
+    }
+    s = "{s};"
+    s
+  }
+}
+
+impl Mutability: ToCode
+impl Var: ToCode {
+  fun toCode(): String { "var" }
+}
+impl Final: ToCode {
+  fun toCode(): String { "final" }
+}
+impl Const: ToCode {
+  fun toCode(): String { "const" }
+}
+
+impl Getter: ToCode {
+  fun toCode(): String {
+    mut let s = "{(type as ToCode).toCode()} get {name}"
+    if body is Some {
+      s = "{s} {(body.unwrap() as ToCode).toCode()}"
+    }
+    if body.map<Bool>({ it is InlineBody }).orElse({ true }) {
+      s = "{s};"
+    }
+    s
+  }
+}
+impl Setter: ToCode {
+  fun toCode(): String {
+    mut let s = "set {name}({(parameter as ToCode).toCode()})"
+    if body is Some {
+      s = "{s} {(body.unwrap() as ToCode).toCode()}"
+    }
+    if body.map<Bool>({ it is InlineBody }).orElse({ true }) {
+      s = "{s};"
+    }
+    s
+  }
+}
+
+impl Body: ToCode
+impl InlineBody: ToCode {
+  fun toCode(): String {
+    "=> {(expression as ToCode).toCode()}"
+  }
+}
+impl Block: ToCode {
+  fun toCode(): String {
+    let statementsCode = (statements as Iterable<Statement>)
+      .map<String>({ (it as ToCode).toCode() })
+      .join("\n")
+    "\👍\n{indent(statementsCode, true)}\n}"
+  }
+}
+
+impl DartType: ToCode
+impl NamedType: ToCode {
+  fun toCode(): String {
+    mut let s = (name as ToCode).toCode()
+    if (typeArguments as Iterable).isNotEmpty() {
+      let generics = (typeArguments as Iterable<DartType>)
+        .map<String>({ (it as ToCode).toCode() })
+        .join(", ")
+      s = "{s}<{generics}>"
+    }
+    s
+  }
+}
+impl FunctionType: ToCode {
+  fun toCode(): String {
+    mut let s = "Function"
+    if _returns is Some {
+      s = "{(_returns.unwrap() as ToCode).toCode()} {s}"
+    }
+    let parameters = parametersToCode(
+      (parameters as Iterable<Parameter>).map<String>({ (it as ToCode).toCode() }),
+      (positionalParameters as Iterable<Parameter>).map<String>({ (it as ToCode).toCode() }),
+      (namedParameters as Iterable<Parameter>).map<String>({ (it as ToCode).toCode() }),
+    )
+    "{s}(parameters)"
+  }
+}
+
+impl Expression: ToCode
+impl Identifier: ToCode {
+  fun toCode(): String {
+    mut let s = name
+    if prefix is Some {
+      print("f")
```

Comment:
Can this be removed?

Reply:
Oh, I was wondering why it prints an `f` 😁